codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ServerDowngrader.java

```diff
@@ -78,9 +80,12 @@ protected void decode(ChannelHandlerContext ctx, Http2StreamFrame frame, List<Ob
                     out.add(full);
                 }
             } else {
-                out.add(HttpConversionUtil.toHttpRequest(id, headersFrame.headers(), validateHeaders));
+                HttpRequest req = HttpConversionUtil.toHttpRequest(id, headersFrame.headers(), validateHeaders);
+                if (!HttpUtil.isContentLengthSet(req)) {
+                    HttpUtil.setTransferEncodingChunked(req, true);
```

Comment:
Hmm, that's not right.  It changes it unconditionally in [setTransferEncodingChunked][0].

```java
    /**
     * Set the {@link HttpHeaderNames#TRANSFER_ENCODING} to either include {@link HttpHeaderValues#CHUNKED} if
     * {@code chunked} is {@code true}, or remove {@link HttpHeaderValues#CHUNKED} if {@code chunked} is {@code false}.
     * @param m The message which contains the headers to modify.
     * @param chunked if {@code true} then include {@link HttpHeaderValues#CHUNKED} in the headers. otherwise remove
     * {@link HttpHeaderValues#CHUNKED} from the headers.
     */
    public static void setTransferEncodingChunked(HttpMessage m, boolean chunked) {
        if (chunked) {
            m.headers().add(HttpHeaderNames.TRANSFER_ENCODING, HttpHeaderValues.CHUNKED);
            m.headers().remove(HttpHeaderNames.CONTENT_LENGTH);
        } else {
```

[0]: https://github.com/netty/netty/blob/4.1/codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java#L302-L313